### PR TITLE
crypto: fix TestToECDSAErrors for nocgo

### PR DESCRIPTION
**Pull request by dindinw**
see https://github.com/ethereum/go-ethereum/issues/15744

### CODE REVIEW

**Comment by GitCop** [comment]
Thank you for your contribution! Your commits seem to not adhere to the repository coding standards

 * Commit: 332931066a41f79ffd3801410365baee08a42afd
  * Commits must be prefixed with the package(s) they modify

Please check the [contribution guidelines](https://github.com/ethereum/go-ethereum#contribution) for more details.

---
This message was auto-generated by https://gitcop.com

File: crypto/crypto.go [hunk 1]
```diff
@@ -97,6 +97,17 @@ func toECDSA(d []byte, strict bool) (*ecdsa.PrivateKey, error) {
...
 	}
 	priv.D = new(big.Int).SetBytes(d)
+
+	// The priv.D must < N
+	if priv.D.Cmp(secp256k1_N) >= 0 {
+		return nil, fmt.Errorf("invalid private key, >=N")
+	}
+	// The priv.D must not be zero or negative.
+	zero := new(big.Int).SetInt64(0)
+	if priv.D.Cmp(zero) <= 0 {
```
Comment (fjl) [inline crypto/crypto.go]: A better way to check this is to use `priv.D.Sign() <= 0`.

**Comment by fjl** [comment]
Thank you so much for looking into this.

**Comment by dindinw** [comment]
@fjl  changed to use `priv.D.Sign()` as your suggestion, please review again. Thanks

**Pull request merged**